lib/recipe/bloc/recipe_state.dart

```diff
@@ -17,4 +17,22 @@ class RecipeState extends Equatable {
 
   @override
   List<Object> get props => [recipes];
+
+  Map<String, dynamic> toMap() {
+    return {
+      'recipes': recipes.map((x) => x.toJson()).toList(),
+    };
+  }
+
+  factory RecipeState.fromMap(Map<String, dynamic> map) {
+    return RecipeState(
+      recipes:
+          List<Recipe>.from(map['recipes']?.map((x) => Recipe.fromJson(x))),
+    );
+  }
+
+  String toJson() => json.encode(toMap());
+
+  factory RecipeState.fromJson(String source) =>
+      RecipeState.fromMap(json.decode(source));
```

Comment:
So this is for serializing all Recipes if I understand correctly. We'll have to think about a consistent implementation across the `Recipe` and `Collecion` modules, but for now it seems good.